controller.py

```diff
@@ -45,9 +48,34 @@ def create_listener(self):
 
 
 if __name__ == '__main__':
+    args = sys.argv[1:]
     logger.info('')
     logger.info('starting logon server')
     logger.info('')
-    c = Controller()
+    if args:
+        # get database configuration info from arguments and password from input.
+        from getpass import getpass
+
+        db_implementation, host, username, db_name, *_ = args
+        password = getpass(prompt='Enter database password: ')
+    else:
+        # load database configuration from params.json file.
+        import json
+
+        with open('./params.json', 'r') as f:
+            data = json.loads(f.read())
+        db_implementation = 'mysql'
+        host = data['db_host']
+        username = data['db_username']
+        password = data['db_password']
+        db_name = data['db_name']
+
+    # Set up the database object based on database implementation selected
+    db = {'mysql': MySQLDB(username, password, host, host, db_name)}[db_implementation]
+
+    # Setup the controller.
+    c = Controller(db)
+
+    # Start the
```

Comment:
The db implementation is now configurable via arguments or loaded from the params.json file as normal if no args are provided. This allows us to create alternative db implementations and use them for a single run without modifying the params file, which could be useful for mocking a db for testing.